docs/docs/reference/release-notes/migrating-from-v2-to-v3.md

```diff
@@ -0,0 +1,848 @@
+
+```
+Should not import the named export 'myProp' from default-exporting module (only default export is available soon)
+```
+
+```diff:title=src/components/Navigation.js
+import React from "react"
+- import { items } from "../data/navigation.json"
++ import navigationData from "../data/navigation.json"
+
+const Navigation = () => (
+  <nav>
+    <ul>
+-      {items.map(item => {
+-        return <li><a href={item.to}>{item.text}</a></li>
+-      })}
++      {navigationData.items.map(item => {
++        return <li><a href={item.to}>{item.text}</a></li>
++      })}
+    </ul>
+  </nav>
+)
+
+export default Navigation
+```
+
+### webpack deprecation messages
+
+When running community Gatsby plugins, you might see `[DEP_WEBPACK]` messages popup during the "Building Javascript" or the "Building SSR bundle" phase. These often mean that the plugin is not compatible with webpack 5 yet. Contact the Gatsby plugin author or the webpack plugin author to flag this issue. Most of the time Gatsby will build fine, however there are cases that it won't and the reasons why could be cryptic.
+
+## Using `fs` in SSR
+
+Gatsby v3 introduce incremental builds for HTML generation. For this feature to work correctly Gatsby needs to track all inputs used to generate HTML file. Arbitrary code execution in `gatsby-ssr.js` files allow usage of `fs` module which is marked as unsafe and result in disabling of this feature. To migrate you can use `import` instead of `fs`:
+
+```diff:title=gatsby-ssr.js
+import * as React from "react"
+-import * as fs from "fs"
+-import * as path from "path"
++import stylesToInline from "!!raw-loader!/some-auto-generated.css"
+
+export function onRenderBody({ setHeadComponents }) {
+-  const stylesToInline = fs.readFileSync(path.join(process.cwd(), `some-auto-generated.css`))
+  setHeadComponents(
+    <style
+      dangerouslySetInnerHTML={{
+        __html: stylesToInline,
+      }}
+    />
+  )
+}
+```
+
+## For Plugin Maintainers
+
+In most cases you won't have to do anything to be v3 compatible, but there are a few things you can do to be certain your plugin won't throw any warnings or errors.
+
+### Setting the proper peer dependencies
+
+`gatsby` should be included under `peerDependencies` of your plugin and it should specify the proper versions of support.
+
+```diff:title=package.json
+{
+  "peerDependencies": {
+-   "gatsby": "^2.32.0",
++   "gatsby": "^3.0.0",
+  }
+}
+```
+
+## Known Issues
+
+This section is a work in progress and will be expanded when necessary. It's a list of known issues you might run into while upgrading Gatsby to v3 and how to solve them.
+
+### `reach-router`
+
+We vendored [reach-router](https://github.com/gatsbyjs/reach-router) to make it work for React 17. We added a webpack alias so that you can continue using it as usual, however you might run into an error like this after upgrading:
```

Comment:
```suggestion
We vendored [reach-router](https://github.com/gatsbyjs/reach-router) to make it work for React 17. We added a webpack alias so that you can continue using it as usual. However, you might run into an error like this after upgrading:
```